spec/models/invite_spec.rb

```diff
@@ -317,6 +317,31 @@
       Invite.redeem_from_email('test2@example.com')
       expect(invite.reload).not_to be_redeemed
     end
+
+    it 'does not work with expired invites' do
+      invite.update!(expires_at: 1.day.ago)
+      Invite.redeem_from_email(user.email)
+      expect(invite).not_to be_redeemed
```

Comment:
```suggestion
      expect(invite.reload).not_to be_redeemed
```

If we do not reload the record, the assertion is just testing the initial state of the invite which is going to always be not be redeemed.

Reply:
Can you please fix the test cases below too.

Reply:
Note the similar pattern just above:

https://github.com/discourse/discourse/blob/3eaac567978026064bbe004cfe1d3dd4cf0091a7/spec/models/invite_spec.rb#L232-L235

Also even without reload the test is failing if the conditions are not met so I'm not sure if reload is required here.

Reply:
Ah I thought `redeemed` is a column on the invite but it isn't. Instead it is calling the `Invite#redeemed?` method which ends up loading an association from the database.